docs/UpgradeGuide.rst

```diff
@@ -119,6 +119,42 @@ If you are using version ``8.0`` of the ``neo4j`` gem, that will be accessible,
 .. code-block:: ruby
 
   Neo4j::ActiveBase.current_session
+  
+ Transactions
+ ^^^^^^^^^^^^
+
+In version ``8.0`` of the ``neo4j`` gem, the recommended API for interacting with transactions has changed. In ``7.0``, the API looked like
+
+.. code-block:: ruby
+
+  Neo4j::Transaction.run do |tx|
+    # do stuff
+    tx.mark_failed
+  end
+  
+In ``8.0``, the API for interacting with transactions looks like
+
+.. code-block:: ruby
+
+  Neo4j::ActiveBase.run_transaction do |tx|
+    # do stuff
+    tx.mark_failed
+  end
```

Comment:
The examples are good, but the first example is a pure `neo4j-core` example, so we should present an the new `neo4j-core` example below first and then say something like "If you are using `ActiveRecord` / `ActiveRel` in the `neo4j` gem, you should use..." and then present this example.

Reply:
Also, since we'd be talking about the `neo4j-core` gem the version change wouldn't be from 7 to 8.  But since this is all under the "The neo4j gem from 7.x to 8.x and the neo4j-core gem from 6.x to 7.x" header we could just skip talking about versions and talk about what's changed